### PR TITLE
Add opt-in toggle for zwave-js telemetry to config panel

### PR DESCRIPTION
**Pull request by cgarwood**
## Proposed change

Add a card with toggle switch for opting into the Z-Wave JS data collection/telemetry. Open to suggestions on better wording for the card. There was some talk on Discord in #devs_zwave about not making it translatable, so I haven't done translations, but happy to change that if desired.

Backend PR has been merged.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by AlCalzone** [comment]
> Open to suggestions on better wording for the card.

Our "template" text is probably a bit long: https://zwave-js.github.io/node-zwave-js/#/data-collection/user-disclosure
Nonetheless, I'd add two things:
1. We only collect software versions and device identifiers
1. The information is important to focus our development efforts and improve the user experience

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts [hunk 1]
```diff
@@ -188,6 +224,14 @@ class ZWaveJSConfigDashboard extends LitElement {
     if (this._status === "connected") {
       this._icon = mdiCheckCircle;
     }
+    const dataCollectionStatus = await fetchDataCollectionStatus(
+      this.hass!,
+      this.configEntryId
+    );
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts:230]: You can do this together with `fetchNetworkStatus`:

```
const [network, dataCollectionStatus] = await Promise.all([fetchNetworkStatus(this.hass!, this.configEntryId), etchDataCollectionStatus(
      this.hass!,
      this.configEntryId
    )]);
```

**Review by bramkragten — APPROVED** [review]

**Comment by raman325** [comment]
Posting here for visibility - there's three states: no preference specified (parameter value of null), opted out (false), and opted in (true). I think we wanted those to be distinct in the UI

**Comment by cgarwood** [comment]
Currently, the toggle switch will always show the current state (if telemetry is no-preference or disabled in HA, but someone uses zwjs2mqtt and enabled it there, the toggle will show on).

Toggling the switch on or off in HA will always set the setting in HA to the state of the switch and enable or disable on the server.

**Comment by raman325** [comment]
> Currently, the toggle switch will always show the current state (if telemetry is no-preference or disabled in HA, but someone uses zwjs2mqtt and enabled it there, the toggle will show on).
> 
> Toggling the switch on or off in HA will always set the setting in HA to the state of the switch and enable or disable on the server.

@MartinHjelmare FYI ^

**Comment by MartinHjelmare** [comment]
That sounds good and in accordance to what we've discussed I think. Right, @raman325?

**Comment by raman325** [comment]
It's slightly different, but arguably better. In the case of using zwavejs2mqtt, I can opt in on the server side and then never provide a preference in HA, and it will always show as enabled because z2m will enable statistics on startup (so the only thing a user can do within HA is opt out). The downside to this approach is that for those users, we will not be able to share the HA version being used

**Comment by raman325** [comment]
> Currently, the toggle switch will always show the current state (if telemetry is no-preference or disabled in HA, but someone uses zwjs2mqtt and enabled it there, the toggle will show on).
> 
> Toggling the switch on or off in HA will always set the setting in HA to the state of the switch and enable or disable on the server.

BTW just rereading this - if the user explicitly opts out, HA will disable statistics on startup, so the flag will show as off.

**Comment by MartinHjelmare** [comment]
If the user opts out and then opts in again, it will activate Home Assistant version reporting, right?

All in all, it's ok I think. Any other thoughts or change suggestions?

**Comment by raman325** [comment]
> If the user opts out and then opts in again, it will activate Home Assistant version reporting, right?
> 
> All in all, it's ok I think. Any other thoughts or change suggestions?

That's correct, and I agree. Just wanted to make sure you saw it since we were discussing it

**Review by raman325 — APPROVED** [review]

**Pull request merged**